frame/asset-conversion/src/lib.rs

```diff
@@ -778,6 +760,13 @@ pub mod pallet {
 					}
 					i.saturating_inc();
 				}
+				Self::deposit_event(Event::SwapExecuted {
```

Comment:
The event emission is part of an `if` now whereas it was emitted previously in either case?  
I hope a test will fail for that :laughing:

Reply:
While it's technically more correct for it to be inside the `if`, it's not currently possible to get to the `else` block: Of the four calls to this function, two construct the path themselves so will always be a length of 2. The other two validate the path length is 2 https://github.com/paritytech/substrate/blob/074dc1dfee4132b597db6eb147a86c72b751f03c/frame/asset-conversion/src/lib.rs#L1070 (i.e. the `if` branch is always taken)

Reply:
I've made it error should the if fail.